shared-module/display/ili9341.c

```diff
@@ -0,0 +1,229 @@
+    write_data(self, 0x00);
+    write_data(self, 0x78);
+
+    write_command(self, ILI9341_PWCTRSEQ);
+    write_data(self, 0x39);
+    write_data(self, 0x2C);
+    write_data(self, 0x00);
+    write_data(self, 0x34);
+    write_data(self, 0x02);
+
+    write_command(self, ILI9341_PUMP);
+    write_data(self, 0x20);
+
+    write_command(self, ILI9341_TIMCTRB);
+    write_data(self, 0x00);
+    write_data(self, 0x00);
+
+    write_command(self, ILI9341_PWCTR1);
+    write_data(self, 0x23);
+
+    write_command(self, ILI9341_PWCTR2);
+    write_data(self, 0x10);
+
+    write_command(self, ILI9341_VMCTR1);
+    write_data(self, 0x3e);
+    write_data(self, 0x28);
+
+    write_command(self, ILI9341_VMCTR2);
+    write_data(self, 0x86);
+
+    write_command(self, ILI9341_MADCTL);
+    write_data(self, 0x48);
+
+    write_command(self, ILI9341_PIXFMT);
+    write_data(self, 0x55);
+
+    write_command(self, ILI9341_FRMCTR1);
+    write_data(self, 0x00);
+    write_data(self, 0x18);
+
+    write_command(self, ILI9341_DFUNCTR);
+    write_data(self, 0x08);
+    write_data(self, 0x82);
+    write_data(self, 0x27);
+
+    write_command(self, ILI9341_ENGMCTR);
+    write_data(self, 0x00);
+
+    write_command(self, ILI9341_GAMMASET);
+    write_data(self, 0x01);
+
+    write_command(self, ILI9341_GMCTRP1);
+    write_data(self, 0x0F);
+    write_data(self, 0x31);
+    write_data(self, 0x2B);
+    write_data(self, 0x0C);
+    write_data(self, 0x0E);
+    write_data(self, 0x08);
+    write_data(self, 0x4E);
+    write_data(self, 0xF1);
+    write_data(self, 0x37);
+    write_data(self, 0x07);
+    write_data(self, 0x10);
+    write_data(self, 0x03);
+    write_data(self, 0x0E);
+    write_data(self, 0x09);
+    write_data(self, 0x00);
+
+    write_command(self, ILI9341_GMCTRN1);
+    write_data(self, 0x00);
+    write_data(self, 0x0E);
+    write_data(self, 0x14);
+    write_data(self, 0x03);
+    write_data(self, 0x11);
+    write_data(self, 0x07);
+    write_data(self, 0x31);
+    write_data(self, 0xC1);
+    write_data(self, 0x48);
+    write_data(self, 0x08);
+    write_data(self, 0x0F);
+    write_data(self, 0x0C);
+    write_data(self, 0x31);
+    write_data(self, 0x36);
+    write_data(self, 0x0F);
+
+    write_command(self, ILI9341_SLPOUT);
+    mp_hal_delay_ms(120);
+
+    write_command(self, ILI9341_DISPON);
+}
+
+void display_ili9341_set_rotation(ili9341_obj_t *self,
+        display_rotation_t rotation)
+{
+    uint8_t data;
+
+    self->rotation = rotation;
+
+    write_command(self, ILI9341_MADCTL);
+
+    switch (self->rotation) {
+        case ROTATION_90:  data = (ILI9341_MADCTL_MV | ILI9341_MADCTL_BGR); break;
+        case ROTATION_180: data = (ILI9341_MADCTL_MY | ILI9341_MADCTL_BGR); break;
+        case ROTATION_270: data = (ILI9341_MADCTL_MX | ILI9341_MADCTL_MY | ILI9341_MADCTL_MV | ILI9341_MADCTL_BGR); break;
+        default:           data = (ILI9341_MADCTL_MX | ILI9341_MADCTL_BGR); break;
+    }
+
+    write_data(self, data);
```

Comment:
What about flip/mirror?

Reply:
Will do also, for now it was not crucial for the RFC but my LCD is rotated 90 degrees so I wanted that :)